src/correlations.jl

```diff
@@ -1,246 +1,186 @@
-export SpectrumSolver, FFTCorrelation, ExponentialSeries
-export correlation_3op_2t, correlation_2op_2t, correlation_2op_1t, spectrum
-
-abstract type SpectrumSolver end
-
-struct FFTCorrelation <: SpectrumSolver end
-
-struct ExponentialSeries{T<:Real,CALC_SS} <: SpectrumSolver
-    tol::T
-    ExponentialSeries(tol::T, calc_steadystate::Bool = false) where {T} = new{T,calc_steadystate}(tol)
+export correlation_3op_2t, correlation_3op_1t, correlation_2op_2t, correlation_2op_1t
+
+function _check_correlation_time_list(tlist::AbstractVector)
+    any(t -> t == 0, tlist) ||
+        throw(ArgumentError("The time list for calculating correlation function must contain the element `0`"))
+    all(>=(0), tlist) ||
+        throw(ArgumentError("All the elements in the time list for calculating correlation function must be positive."))
+    return nothing
 end
 
-ExponentialSeries(; tol = 1e-14, calc_steadystate = false) = ExponentialSeries(tol, calc_steadystate)
-
 @doc raw"""
-    correlation_3op_2t(H::QuantumObject,
-        ψ0::QuantumObject,
-        t_l::AbstractVector,
-        τ_l::AbstractVector,
+    correlation_3op_2t(H::AbstractQuantumObject,
+        ψ0::Union{Nothing,QuantumObject},
+        tlist::AbstractVector,
+        τlist::AbstractVector,
+        c_ops::Union{Nothing,AbstractVector,Tuple},
```

Comment:
Here we force the users to insert the collapse operators. I think that this should left free, as one can in principle calculate the correlation function also for the closed system case. In think that qutip also supports this.

Reply:
just specify `nothing` or empty `c_ops` list.

QuTiP can only specify empty `c_ops` for this situation.

Reply:
Okok fine